airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java

```diff
@@ -27,7 +29,20 @@ public class AdvancedTestDataComparator implements TestDataComparator {
   public static final String AIRBYTE_DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
   public static final String AIRBYTE_DATETIME_PARSED_FORMAT = "yyyy-MM-dd HH:mm:ss.S";
   public static final String AIRBYTE_DATETIME_PARSED_FORMAT_TZ = "yyyy-MM-dd HH:mm:ss XXX";
-  public static final String AIRBYTE_DATETIME_WITH_TZ_FORMAT = "yyyy-MM-dd'T'HH:mm:ssXXX";
+  public static final String AIRBYTE_DATETIME_WITH_TZ_FORMAT = "[yyyy][yy]['-']['/']['.'][' '][MMM][MM][M]['-']['/']['.'][' '][dd][d]"
+      + "[[' ']['T']HH:mm[':'ss[.][SSSSSS][SSSSS][SSSS][SSS][' '][z][zzz][Z][O][x][XXX][XX][X][' '][G]]]";
+
+  // TODO revisit dataset which used date as string: exchange_rate_catalog.json
+  // tried to change it to date time type but some connectors failed to store it e.i.
+  // bigquery-denormalized
```

Comment:
does this comment imply that if we merge this change to master, we will see new failures when bigquery-denormalized is trying to process datetime values? Or will this still be OK until bigquery-denormalized declares that it supports v1 protocol?

Reply:
Yeah, there were two issues here. 
1. Previous test comparison had a wrong regex to find date time with Tz values:
```
  protected boolean isDateTimeWithTzValue(final String value) {
    return value.matches(".+[+-]\\d{2}:\\d{2}");
  }
```
I had to fix it to handle it for avro values (it used microseconds). 
2. After this fix I got couple of dates which schema defined as a string but not date (dataset V0, V1). Tried to update schema  to: 
```
              "type": "string",
              "format": "date-time",
              "airbyte_type": "timestamp_with_timezone"
```
but  bigquery-denormalized failed to store `2020-08-31T00:00:00Z`.  I was not sure how this change affect tests for others connectors also, so created this workaround 

Created the issue https://github.com/airbytehq/airbyte/issues/20362. Don't have a logs now but will try to extend with details